File: experiments/engine-kt/adapter-openai-responses/src/main/kotlin/com/garive/eng/kt/openai/Wire.kt
package com.garive.eng.kt.openai

/** Internal Responses HTTP vocabulary. */
internal object HttpWire {
    const val METHOD_POST: String = "POST"
    const val HEADER_CONTENT_TYPE: String = "content-type"
    const val HEADER_ACCEPT: String = "accept"
    const val MEDIA_JSON: String = "application/json"
    const val MEDIA_SSE: String = "text/event-stream"
}

/** Internal Responses JSON field vocabulary used across codecs. */
internal object ResponseFields {
    const val TYPE: String = "type"
    const val RESPONSE: String = "response"
    const val ERROR: String = "error"
    const val ROLE: String = "role"
    const val SEQUENCE_NUMBER: String = "sequence_number"

    val CREATE: Set<String> = setOf(
        "model", "input", "stream", "max_output_tokens", "temperature", "top_p",
        "truncation", "tools", "tool_choice", "parallel_tool_calls", "text",
        "reasoning", "metadata", "stream_options",
    )
}

/** Internal Responses discriminators shared by parsing and lifecycle checks. */
internal object ResponseKinds {
    const val RESPONSE: String = "response"
    const val MESSAGE: String = "message"
    const val FUNCTION_CALL: String = "function_call"
    const val REASONING: String = "reasoning"
    const val OUTPUT_TEXT: String = "output_text"
    const val REFUSAL: String = "refusal"
    const val ASSISTANT: String = "assistant"
    const val FUNCTION_CALL_OUTPUT: String = "function_call_output"
    const val INPUT_TEXT: String = "input_text"
    const val INPUT_IMAGE: String = "input_image"
    const val FUNCTION: String = "function"
    const val TEXT: String = "text"
    const val JSON_OBJECT: String = "json_object"
    const val JSON_SCHEMA: String = "json_schema"
}

File: adapters/openai-responses/src/wire.rs
//! Internal wire vocabulary shared by the Responses codecs.

pub(crate) const METHOD_POST: &str = "POST";
pub(crate) const HEADER_CONTENT_TYPE: &str = "content-type";
pub(crate) const HEADER_ACCEPT: &str = "accept";
pub(crate) const MEDIA_JSON: &str = "application/json";
pub(crate) const MEDIA_SSE: &str = "text/event-stream";

pub(crate) const FIELD_TYPE: &str = "type";
pub(crate) const FIELD_RESPONSE: &str = "response";
pub(crate) const FIELD_SEQUENCE_NUMBER: &str = "sequence_number";

pub(crate) const KIND_MESSAGE: &str = "message";
pub(crate) const KIND_FUNCTION_CALL: &str = "function_call";
pub(crate) const KIND_REASONING: &str = "reasoning";
pub(crate) const KIND_OUTPUT_TEXT: &str = "output_text";
pub(crate) const KIND_REFUSAL: &str = "refusal";
pub(crate) const CREATE_FIELDS: &[&str] = &[
    "model",
    "input",
    "max_output_tokens",
    "temperature",
    "top_p",
    "truncation",
    "tools",
    "tool_choice",
    "parallel_tool_calls",
    "text",
    "reasoning",
    "metadata",
    "stream",
    "stream_options",
];
